### PR TITLE
Add REST API `Cost Estimate`; Configure JavaScript Mode to use ECMAScript 2021 (ES12)

### PR DESCRIPTION
**Pull request by jasonburkert-ccpo**
## Summary
- [Configures JavaScript Mode](https://developer.servicenow.com/blog.do?p=/post/tokyo-ecmascript-2021/#how) of ATAT scoped application to use `ECMAScript 2021 (ES12)` rather than `ES5 Standards Mode`. This is only recently available to us since to environment upgrade to Tokyo version.  Unlocks [these language features](https://www.servicenow.com/community/developer-articles/master-ecmascript-a-comprehensive-list-with-real-world-examples/ta-p/2420283) for our use.
- Adds REST API `Cost Estimate`
  - Adds resource `Get Data` - to expose data for use on Summary UI page
- Adds method `Igce.getCostEstimateData()` which is the bulk of the code the web service wraps

## Details
Commit f5a24ee4f07837f13bf5c6e8a69ac5e1d19dbc2b is largely code by @jeffsegal-ccpo and makes the response of `Igce.getCostEstimateData()` dynamic.  Leverages language features enabled by the JavaScript Mode update such as `Array.prototype.flatMap()`

Surge capability is optional.  When not included it will not be present in the response of the `Get Data` resource.  Likewise, it is also omitted from calculation of the grand total.

### Changes to the IGCE Payload
DITCO's fee percentage was added to IGCE payload whereas before it was omitted and included downstream for IGCE generation (requires follow-on PR, see [this comment](https://github.com/dod-ccpo/atat-snow/pull/347#discussion_r1170649648))
  - Adds `fee` to the `contractingShop` object when shop is _DITCO_

Changed the values of some properties to _NOT USED_ for clarity.  Original IGCE doc had several related fields and we consolidated our use into one.  These are not used in our implementation.  Outright removal may cause shape validation errors.
  - `assumptionsMade` and `informationSource` of the `instructions` object

Ticket: AT-8700

### CODE REVIEW

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml [hunk 1]
```diff
@@ -917,6 +917,78 @@ Igce.prototype = {
         }
     },
 
+    getCostEstimateData: function(packageSysId) {
+        try {
+            var packageId = packageSysId ? packageSysId : this.acquisitionPackage.sys_id;
+            var sampleData = {
```
Comment (jasonburkert-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml:923]: Static JSON returned

File: f600233d1b154d507b782f84604bcb12/update/sys_ws_definition_5a663ee697352110cf3cfd9fe153afe4.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    <sys_ws_definition action="INSERT_OR_UPDATE">
+        <active>true</active>
+        <base_uri>/api/x_g_dis_atat/cost_estimate</base_uri>
+        <consumes>application/json,application/xml,text/xml</consumes>
+        <consumes_customized>false</consumes_customized>
+        <default_version>No active default version</default_version>
+        <doc_link/>
+        <enforce_acl>cf9d01d3e73003009d6247e603f6a990</enforce_acl>
+        <is_versioned>false</is_versioned>
+        <name>Cost Estimate</name>
```
Comment (jasonburkert-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_ws_definition_5a663ee697352110cf3cfd9fe153afe4.xml:11]: new REST API

File: f600233d1b154d507b782f84604bcb12/update/sys_ws_operation_62d293e297b52110cf3cfd9fe153af67.xml [hunk 3]
```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="UTF-8"?><record_update table="sys_ws_operation">
+    <sys_ws_operation action="INSERT_OR_UPDATE">
+        <active>true</active>
+        <consumes>application/json,application/xml,text/xml</consumes>
+        <consumes_customized>false</consumes_customized>
+        <default_operation_uri/>
+        <enforce_acl>cf9d01d3e73003009d6247e603f6a990</enforce_acl>
+        <http_method>GET</http_method>
+        <name>Get Data</name>
```
Comment (jasonburkert-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_ws_operation_62d293e297b52110cf3cfd9fe153af67.xml:9]: new Resource

**Comment by jasonburkert-ccpo** [comment]
## Change to JavaScript Mode of ATAT scoped application

### Old config

image

### New config

image

File: f600233d1b154d507b782f84604bcb12/sys_app_f600233d1b154d507b782f84604bcb12.xml [hunk 4]
```diff
@@ -6,7 +6,7 @@
         <guided_setup_guid/>
         <hide_on_ui>false</hide_on_ui>
         <installed_as_dependency>false</installed_as_dependency>
-        <js_level>helsinki_es5</js_level>
+        <js_level>es_latest</js_level>
```
Comment (jasonburkert-ccpo) [inline f600233d1b154d507b782f84604bcb12/sys_app_f600233d1b154d507b782f84604bcb12.xml:9]: JavaScript Mode update

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml [hunk 5]
```diff
@@ -9,1186 +9,1515 @@
...
+            // TODO Improve to handle DITCO vs external agency and lookup specific fee %; assumes external and 1%
+            var fees = {
+                "CLIN Type & Classification": "External ordering agency fee (1%)",
+            };
+            Object.assign(
+                fees,
+                Object.keys(subtotals).reduce(function(result, key) {
+                    result[key] = subtotals[key] * 0.01;
+                    return result;
+                }, {})
```
Comment (jasonburkert-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml:1008]: Improvement to be made

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml [hunk 6]
```diff
@@ -9,1186 +9,1515 @@
...
+            );
+            fees.Total = sum(fees);
+            data.push(fees);
+
+            // TODO Improve to lookup specific surge %; assumes 5%
+            var surge = Object.keys(subtotals).reduce(function(result, key) {
+                result[key] = subtotals[key] * 0.05;
```
Comment (jasonburkert-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml:1015]: Improvement to be made

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml (hunk 5 above)
Comment (jasonburkert-ccpo) [reply f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml]: Fixed in https://github.com/dod-ccpo/atat-snow/pull/347/commits/3fa8bf5582ab2d8c1abf690f68cb051ef7e09619

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml (hunk 6 above)
Comment (jasonburkert-ccpo) [reply f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml]: Fixed in https://github.com/dod-ccpo/atat-snow/pull/347/commits/95f3b3c8a7f73596ee76be9ea3ef0e2f1204f23f

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml [hunk 7]
```diff
@@ -5,1190 +5,1527 @@
...
+                    }, 0);
+                return periodResult;
+            }, {});
+
+            // Calculation for external ordering fee whether DITCO and OTHER
+
+            // TODO: fix the label below to respect DITCO/OTHER
+            var fees = {
+                "CLIN Type & Classification": "External ordering agency fee (" + payload.templatePayload.contractingShop.fee + "%)",
+            };
```
Comment (jasonburkert-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml:1001]: One last thing to fix?  Need the expected output for both cases.

Comment (jasonburkert-ccpo) [reply f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml]: Fixed in https://github.com/dod-ccpo/atat-snow/pull/347/commits/7e94f9e63318773e90a3205b498c08683b812e03

Followed naming [implemented here](https://github.com/dod-ccpo/atat-web-api/blob/89c9ceb57553e681cdd045b6dfefa77309f6fa1d/document-generation/igce-document.ts#L251)

**Comment by jasonburkert-ccpo** [comment]
## Sample execution of new REST API and response

image

```
{ "result": { "packageId": "f5571908db1e6510b1227ea5f39619ec", "payload": "{\"data\":[{\"CLIN Type & Classification\":\"x002/x018 Cloud Support UNCLASSIFIED\",\"Base Period\":19091,\"Option 1\":9602,\"Option 2\":11313,\"Option 3\":22624,\"Option 4\":22735,\"Total\":85365},{\"CLIN Type & Classification\":\"x004/x020 Cloud Support SECRET CLASSIFIED\",\"Base Period\":900,\"Option 1\":900,\"Option 2\":900,\"Option 3\":900,\"Option 4\":900,\"Total\":4500},{\"CLIN Type & Classification\":\"x006/x022 Cloud Support TOP SECRET CLASSIFIED\",\"Base Period\":20,\"Option 1\":40,\"Option 2\":60,\"Option 3\":80,\"Option 4\":100,\"Total\":300},{\"CLIN Type & Classification\":\"x007 Travel\",\"Base Period\":17000,\"Option 1\":17000,\"Option 2\":17000,\"Option 3\":17000,\"Option 4\":17000,\"Total\":85000},{\"CLIN Type & Classification\":\"x005/x021 Cloud TOP SECRET CLASSIFIED\",\"Base Period\":9590,\"Option 1\":4736,\"Option 2\":5545,\"Option 3\":1740,\"Option 4\":1740,\"Total\":23351},{\"CLIN Type & Classification\":\"x003/x019 Cloud SECRET CLASSIFIED\",\"Base Period\":100,\"Option 1\":1088.88,\"Option 2\":1411.1,\"Option 3\":5872,\"Option 4\":5972,\"Total\":14443.98},{\"CLIN Type & Classification\":\"x001/x017 Cloud UNCLASSIFIED\",\"Base Period\":1341.1,\"Option 1\":563,\"Option 2\":665,\"Option 3\":1752,\"Option 4\":1659,\"Total\":5980.1},{\"CLIN Type & Classification\":\"Contracting Office Fee (6%)\",\"Base Period\":2882.526,\"Option 1\":2035.7927999999997,\"Option 2\":2213.6459999999997,\"Option 3\":2998.08,\"Option 4\":3006.3599999999997,\"Total\":13136.4048}],\"subtotal\":{\"Base Period\":48042.1,\"Option 1\":33929.88,\"Option 2\":36894.1,\"Option 3\":49968,\"Option 4\":50106},\"surge\":{\"Base Period\":1921.684,\"Option 1\":1357.1951999999999,\"Option 2\":1475.764,\"Option 3\":1998.72,\"Option 4\":2004.24},\"total\":{\"Base Period\":52846.31,\"Option 1\":37322.868,\"Option 2\":40583.51,\"Option 3\":54964.8,\"Option 4\":55116.6,\"Total\":240834.08800000002}}" } }
```

**Review by jeffsegal-ccpo — APPROVED** [review]

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml (hunk 5 above)
Comment (jasonburkert-ccpo) [reply f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml]: The above was done in https://github.com/dod-ccpo/atat-web-api/pull/1614

**Pull request merged**